nettests/nettests.go

```diff
@@ -67,252 +74,148 @@ func (c *Controller) SetNettestIndex(i, n int) {
 	c.ntIndex = i
 }
 
-// Init should be called once to initialise the nettest
-func (c *Controller) Init(nt *mk.Nettest) error {
-	log.Debugf("Init: %v", nt)
-	err := c.Ctx.MaybeLocationLookup()
-	if err != nil {
-		return err
-	}
+// Run runs the selected nettest using the related experiment
+// with the specified inputs.
+//
+// This function will continue to run in most cases but will
+// immediately halt if something's wrong with the file system.
+func (c *Controller) Run(exp *experiment.Experiment, inputs []string) error {
+	ctx := context.Background()
```

Comment:
Maybe we can call this `background` or something similar to avoid confusing it with `c.Ctx` (we ought to eventually rename `c.Ctx` to something better)?

Reply:
I'd rather keep it named `ctx` and perform the `c.Ctx` renaming as a follow up. We discussed calling it `Probe`, and hence `probe` when instantiated. I wanted, at the same time, to avoid starting a refactoring such as that one, which is mainly mechanic, with this PR pending.

Reply:
(Just double checked that we've a tracking issue: https://github.com/ooni/probe-cli/issues/45)